pkg/operator/apis/monitoring/v1/http_types_test.go

```diff
@@ -0,0 +1,137 @@
+package v1
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/prometheus/prometheus/secrets"
+	"github.com/prometheus/prometheus/secrets/kubernetes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestClusterSecretKeySelector_toPrometheusSecretRef_PodMonitoring(t *testing.T) {
+	p := &PodMonitoring{
+		ObjectMeta: metav1.ObjectMeta{Namespace: "foo"},
+	}
+
+	t.Run("nil", func(t *testing.T) {
+		pool := PrometheusSecretConfigs{}
+		var c *ClusterSecretKeySelector
+
+		ref, err := c.toPrometheusSecretRef(p, pool)
+		if err != nil {
+			t.Fatalf("unexpected failure: %s", err)
+		}
+		if ref != "" {
+			t.Fatalf("expected empty ref, got %v", ref)
+		}
+		if len(pool.SecretConfigs()) != 0 {
+			t.Fatalf("expected no configs, got %v", pool.SecretConfigs())
+		}
+	})
+
+	// Empty or correct namespace.
+	for _, ns := range []string{"", p.Namespace} {
+		t.Run(fmt.Sprintf("namespace=%v", ns), func(t *testing.T) {
+			pool := PrometheusSecretConfigs{}
+			c := &ClusterSecretKeySelector{
+				Name:      "secret1",
+				Key:       "key1",
+				Namespace: ns,
+			}
+
+			ref, err := c.toPrometheusSecretRef(p, pool)
+			if err != nil {
+				t.Fatalf("unexpected failure: %s", err)
+			}
+			if exp, got := "foo/secret1/key1", ref; exp != got {
+				t.Fatalf("expected ref %v, got %v", exp, got)
+			}
+			if exp, got := []secrets.SecretConfig[kubernetes.SecretConfig]{
+				{
+					Name: "foo/secret1/key1", Config: kubernetes.SecretConfig{
+						Name:      c.Name,
+						Key:       c.Key,
+						Namespace: p.Namespace,
+					},
+				},
+			}, pool.SecretConfigs(); cmp.Diff(exp, got) != "" {
```

Comment:
nit: can you store the result of `cmp.Diff` in a variable and reuse it or does Go not let you in the middle of the conditional?

Reply:
Yup, chaining is bit limited, so can't do this.

![image](https://github.com/GoogleCloudPlatform/prometheus-engine/assets/6950331/0977ba8e-1928-430f-92a9-f5faf57fb1f9)